src/libmongoc/examples/compile-with-pkg-config.sh

```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
 
 # -- sphinx-include-start --
-gcc -o hello_mongoc hello_mongoc.c $(pkg-config --libs --cflags libmongoc-1.0)
```

Comment:
Suggest also documenting the file renames in NEWS. (Can possibly copy and reformat bits from the PR description).